src/Razor/src/Microsoft.VisualStudio.RazorExtension/RazorPackage.cs

```diff
@@ -4,21 +4,32 @@
 using System;
 using System.Runtime.InteropServices;
 using System.Threading;
+using Microsoft.VisualStudio.LanguageServerClient.Razor;
 using Microsoft.VisualStudio.Shell;
 using Task = System.Threading.Tasks.Task;
 
 namespace Microsoft.VisualStudio.RazorExtension
 {
-    [PackageRegistration(UseManagedResourcesOnly = true, AllowsBackgroundLoading = true)]
+    // We attach to the 51st priority order because the traditional Web + XML editors have priority 50. We need to be loaded prior to them
+    // since we want to have the option to own the experience for Razor files
+    [ProvideEditorExtension(typeof(RazorEditorFactory), RazorLSPContentTypeDefinition.CSHTMLFileExtension, 52, NameResourceID = 101)]
+    [ProvideEditorExtension(typeof(RazorEditorFactory), RazorLSPContentTypeDefinition.RazorFileExtension, 52, NameResourceID = 101)]
+    [ProvideEditorFactory(typeof(RazorEditorFactory), 101)]
+    [PackageRegistration(UseManagedResourcesOnly = true)]
```

Comment:
What might an extension for which this is false look like?

Reply:
No clue to be honest. This is part of the default VS extension setup that's recommended.

Reply:
I'd imagine that if you're doing native interop this would be set to false but it's totally a guess 😄